src/vulkan/buffer_descriptor.cc

```diff
@@ -121,6 +121,8 @@ Result BufferDescriptor::CreateOrResizeIfNeeded(
 
     new_buffer->CopyFromBuffer(command, *buffer_);
 
+    not_destroyed_buffers_.push_back(std::move(buffer_));
```

Comment:
Isn't it better to do buffer_->Shutdown() here instead?

Reply:
The problem is that we must copy the content of `buffer_` to `new_buffer`.
The above line `new_buffer->CopyFromBuffer(command, *buffer_);` record a command for the copy but it did not submit the command yet.
If we destroy `buffer_` here, we lost the information when it actually submit the copy command.

An alternative is plumbing vulkan command buffer here and call submission and reset.

Do u think the alternative is better than keeping those buffers in `not_destroyed_buffers_`?

Reply:
Ah, right you have to keep the data.  :-)

I still think it's better to destroy the buffer as soon as you can. So I recommend:

   resize is {
     create a new buffer with the bigger size
     copy the data from the old buffer to the new buffer
     destroy the old buffer
     set buffer_ to the newly allocated buffer.
  }

Reply:
As open a bug #200, I will create a PR to replace it with allocation of maximum VkBuffer before starting the execution instead of resizing it.

Reply:
We changed the design for this issue. We dropped #200 and will go for #207.